analyzer/codechecker_analyzer/cmd/analyze.py

```diff
@@ -190,16 +190,20 @@ def add_arguments_to_parser(parser):
                         dest="report_hash",
                         default=argparse.SUPPRESS,
                         required=False,
-                        choices=['context-free'],
+                        choices=['context-free', 'context-free-v2'],
```

Comment:
I think restructuring the help message would be better here to understand the differences between the values better.
I'm not sure mixing the `if not set` case here is good or not it could be a bit confusing or only mention it at the end.

I think first we should describe the available options and optionally the value if the argument is not set:
```
- context-free ... there was a bug and for clang tidy not the context free hash was generated (kept for backward compatibility)
- context-free-v2 context free hash is used for clangsa and clang-tidy

If the argument is not set context sensitive hash will be generated for clang and clang-tidy.
```

I think structuring the values like this makes them easier to understand.